test/python/circuit/classical/test_expr_constructors.py

```diff
@@ -149,6 +160,11 @@ def test_logic_not_explicit(self):
             expr.Unary(expr.Unary.Op.LOGIC_NOT, expr.Var(clbit, types.Bool()), types.Bool()),
         )
 
+    @ddt.data(expr.logic_not)
+    def test_urnary_logical_forbidden(self, function):
+        with self.assertRaisesRegex(TypeError, "cannot apply"):
+            function(7.0)
```

Comment:
not sure the ddt parametrisation is buying anything here haha

(also: urnary again)